wit/request-reply.wit

```diff
@@ -0,0 +1,16 @@
+/// The request-reply interface allows a guest to send a message and await a response. This
+/// interface is considered optional as not all message services support the concept of
+/// request/reply. However, request/reply is a very common pattern in messaging and as such, we have
+/// included it as a core interface.
+interface request-reply {
+    use types.{client, message, error};
+
+    /// Performs a blocking request/reply operation with an optional timeout. If the timeout value
+    /// is not set, then the request/reply operation will block indefinitely.
+    request: func(c: client, msg: message, timeout-ms: option<u32>) -> result<option<list<message>>, error>;
+
+    /// Replies to the given message with the given response message. The details of which channel 
+    /// the message is sent to is up to the implementation. This allows for reply to details to be
+    /// handled in the best way possible for the underlying messaging system.
+    reply: func(reply-to: borrow<message>, reply: message) -> result<_, error>;
```

Comment:
```suggestion
    reply: func(reply-to: borrow<message>, reply: message, opts: option<reply-options>) -> result<_, error>;
```

```wit
record reply-options {
  timeout-ms: option<u32>
}
```

Same over here. How do we deal with the timeout config?

Reply:
A timeout would only matter here if you're expecting the thing you sent the reply to respond with a confirmation. More complex patterns like what Kafka Streams and NATS JetStream provide are going to be layered on top of this. Those types of patterns can handle things like acks and double_acks. Everything in this package, with the exception of request reply (where the request should have an eventual timeout), is doing basic pub sub with no explicit guarantees around deliverability

Reply:
> A timeout would only matter here if you're expecting the thing you sent the reply to respond with a confirmation.

When you say "reply to respond with a confirmation," what do you have in mind?

I am focused more on network timeout; the guarantee is about the system, at the very least, getting the message.